drop-transfer/src/transfer.rs

```diff
@@ -16,15 +16,15 @@ pub struct Transfer {
 }
 
 impl Transfer {
-    pub fn new(peer: IpAddr, files: Vec<File>, config: &DropConfig) -> Result<Self> {
+    pub fn new(peer: IpAddr, files: Vec<File>, config: Arc<DropConfig>) -> Result<Self> {
         Self::new_with_uuid(peer, files, Uuid::new_v4(), config)
     }
 
     pub(crate) fn new_with_uuid(
         peer: IpAddr,
         files: Vec<File>,
         uuid: Uuid,
-        config: &DropConfig,
+        config: Arc<DropConfig>,
```

Comment:
Um, why? We don't need ownership of config here, plain reference suffices

Reply:
I changed DropConfig to be wrapped inside an Arc and fixed a ton of issues surrounding that so it simply slipped here and here. Anyway removed it